server/controllers/patients/patchPatientData.js

```diff
@@ -0,0 +1,59 @@
+const {
+  patchPatientDataByIdQuery,
+  patientCheckPhone,
+} = require('../../database/queries');
+const { boomify, patientDataValidation } = require('../../utils');
+
+const patchPatientData = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      phone,
+      email,
+      birthday,
+      diseases,
+      patientId,
+    } = await patientDataValidation.validate(
+      { ...req.body, ...req.params },
+      {
+        abortEarly: false,
+      },
+    );
+
+    const {
+      rows: [patient],
+    } = await patientCheckPhone({ phone });
+    if (patient) {
+      return next(boomify(409, 'Edit Error', 'phone number is exist'));
+    }
+
+    const {
+      rows: [data],
+    } = await patchPatientDataByIdQuery({
+      firstName,
+      lastName,
+      phone,
+      email,
+      birthday,
+      diseases,
+      patientId,
+    });
+
+    return res.json({
+      statusCode: 200,
+      message: 'Updated successfully',
+      data,
+    });
```

Comment:
If you tried to edit an Id that doesn't exist you will always receive 200 status code and `Updated successfully`
```suggestion
    if (rowCount === 1) {
      return res.json({
        statusCode: 200,
        message: 'Updated successfully',
        data,
      });
    }
    return next(boomify(404, `There's no patient with this Id`));
```

Reply:
I liked this one and how to give attention to details :art: